### ISSUE TITLE
constantly re-creating new Module eventually runs out of wasm memory, what's going on? i thought the memory was garbage collected, but it didn't

### ISSUE THREAD

**Issue by Ma-Y** [emscripten-core/emscripten#19008]
latest version

error message:
out of memory, failed at line ``` new Webassembly.memory....```

under windows task manager, when creating a new Module , constantly adds 100M more memory usage, why is it not getting garbage collected

can i only create them but no way to deallocate the wasm memory?

by the way , i also reassured that the old Module is no longer used, it's not running an infinite loop, i also assigned the module to be "module = null", i dont know if it's because of i enabled -ASYNCIFY , but i tried my best to keep no reference to old Module, to make garbage collector happy to collect it.

my question is , why garbage collector isn't collecting the old new Webassembly.memory, is there any other reference to it? or is it simply can't because it only able to allocate new one but not cleaning them since this kind of memory are special?

Update:
so basically I have multiple modules running, but since i can't first load them all, and let them sleep on some conditional variables, I decide to re-create a new one each time I needed it, so I try to make sure that previous module's code is finished, by that I mean main() is finished, then I point the reference to nothing ```null```, by doing this I natively thought I was doing things to free it, but turns out it's continuously consuming memory every time it creates a new Module, Plz help!! guys.

**Comment by Ma-Y** [comment]
by the way , i also reassured that the old Module is no longer used, it's not running an infinite loop, i also assigned the module to be "module = null", i dont know if it's because of i enabled -ASYNCIFY , but i tried my best to keep no reference to old Module, to make garbage collector happy to collect it

**Comment by Ma-Y** [comment]
i think i found the solution, I 'll simply create the module constant amount of times. since variable amount of time will have this issue.
